scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/signal/SignalApiImpl.java

```diff
@@ -108,17 +109,26 @@ public boolean isReceived(String signalName) throws SignalApiException {
     public String checkForSignals(Set<String> signalsSubSet) throws SignalApiException {
         init();
         HashSet<String> signals = getJobSignals();
+        String receivedString = null;
         if (!signals.isEmpty()) {
-            return signalsSubSet.stream().filter(signals::contains).findFirst().orElse(null);
-        } else {
-            return null;
+            receivedString = signalsSubSet.stream().filter(signals::contains).findFirst().get();
+            // Remove ready signals if a signal is received
+            if (receivedString != null) {
+                removeManySignals(signalsSubSet.stream()
```

Comment:
we mentioned that for this case, it is the template workflow which should remove the signals, not the api.